exonum_derive/src/tx_set.rs

```diff
@@ -0,0 +1,171 @@
+// Copyright 2018 The Exonum Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use proc_macro::TokenStream;
+use proc_macro2::{Ident, Span};
+use syn::{Data, DataEnum, DeriveInput, Type};
+
+struct TxSetVariant(u16, Ident, Type);
+
+fn get_tx_variants(data: &DataEnum) -> Vec<TxSetVariant> {
+    if data.variants.is_empty() {
+        panic!("TransactionSet enum should not be empty");
+    }
+
+    data.variants
+        .iter()
+        .enumerate()
+        .map(|(n, v)| {
+            if v.fields.iter().len() > 1 {
```

Comment:
- IMO, it would make sense to extract `v.fields.iter()` into a variable.
- It seems that we don't support named fields in variants, but this code seemingly doesn't check this.